carp.studies.core/src/commonTest/kotlin/dk/cachet/carp/studies/domain/users/RecruitmentTest.kt

```diff
@@ -123,20 +128,34 @@ class RecruitmentTest
 
         assertFalse( recruitment.getStatus() is RecruitmentStatus.ReadyForDeployment )
 
-        val participantIds = setOf( participant.id )
-        assertFailsWith<IllegalStateException> { recruitment.addParticipantGroup( participantIds ) }
+        val roleAssignment = setOf( AssignedParticipantRoles( participant.id, AssignedTo.All ) )
+        assertFailsWith<IllegalStateException> { recruitment.addParticipantGroup( roleAssignment ) }
         val participationEvents = recruitment.consumeEvents().filterIsInstance<Recruitment.Event.ParticipantGroupAdded>()
         assertEquals( 0, participationEvents.count() )
     }
 
+    @Test
+    fun addParticipantGroup_fails_when_role_not_configured_in_protocol()
+    {
+        val recruitment = Recruitment( studyId )
+        val participant = recruitment.addParticipant( participantEmail )
+        val protocol = createEmptyProtocol()
+        recruitment.lockInStudy( protocol.getSnapshot(), StudyInvitation( "Some study" ) )
+        val role1 = ParticipantRole( "Role 1", isOptional = false )
```

Comment:
It would help readability to call this `unknownRole`, or `nonExistingRole`. The subsequent variables can be shorter, or even inlined. They matter less.